lib/puma/server.rb

```diff
@@ -896,11 +900,16 @@ def notify_safely(message)
       begin
         @notify << message
       rescue IOError
+        Thread.current.purge_interrupt_queue if Thread.current.respond_to? :purge_interrupt_queue
         # The server, in another thread, is shutting down
       rescue RuntimeError => e
         # The server, in another thread, has been shut down during the system call
         # https://github.com/puma/puma/pull/1206
-        raise e unless e.message.include?('IOError')
+        if e.message.include?('IOError')
```

Comment:
> with no real purpose

I read the source code and my first thought was: why this strange way to detect IOError. Then I checked code above and saw `rescue IOError`. I got totally puzzled. I opened https://github.com/puma/puma/pull/1206, and only there I found that message reads as "can't modify frozen IOError"